lib/realtime/vehicle_or_ghost.ex

```diff
@@ -4,59 +4,87 @@ defmodule Realtime.VehicleOrGhost do
   @type t :: Vehicle.t() | Ghost.t()
 
   @spec find_by([t()], Server.search_params()) :: [t()]
-  def find_by(vehicles, %{
-        text: text,
-        property: :all
-      }),
-      do:
-        filter_by_prop_matching(
-          vehicles,
-          [:run_id, :label, :operator_id, :operator_first_name, :operator_last_name],
-          text
-        )
-
-  def find_by(vehicles, %{
-        text: text,
-        property: :run
-      }),
-      do: filter_by_prop_matching(vehicles, :run_id, text)
-
-  def find_by(vehicles, %{
-        text: text,
-        property: :vehicle
-      }),
-      do: filter_by_prop_matching(vehicles, :label, text)
-
-  def find_by(vehicles, %{
-        text: text,
-        property: :operator
-      }),
-      do:
-        filter_by_prop_matching(
-          vehicles,
-          [:operator_id, :operator_first_name, :operator_last_name],
-          text
-        )
-
-  @spec filter_by_prop_matching([t()], atom() | [atom()], String.t()) :: [t()]
+  def find_by(vehicles, %{text: text, property: search_property, limit: limit}) do
+    search_terms = text |> clean_for_matching |> Enum.reject(&(String.length(&1) < 2))
+
+    if search_terms == [] do
+      []
+    else
+      take_by_props_matching(
+        vehicles,
```

Comment:
this list of vehicles should be sorted by relevance first (logged on time), but this PR is getting meaty. Will create a separate PR for that logic.